core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider.go

```diff
@@ -124,6 +125,7 @@ func (p *logEventProvider) Start(context.Context) error {
 
 	p.lggr.Infow("starting log event provider", "readInterval", p.opts.ReadInterval, "readMaxBatchSize", readMaxBatchSize, "readers", readerThreads)
 
+	p.routines.Add(readerThreads)
 	{ // start readers
 		go func(ctx context.Context) {
```

Comment:
Just a side node but why is this go routine needed? We're starting the reader in separate go routines inside this function anyway?